portfolio/src/main/webapp/script.js

```diff
@@ -50,16 +50,28 @@ function showPics() {
 }
 
 /* Gets array list of messages from servlet, converts to JSON, and displays it on home page. */
-function getJSONMessages() {
+function getBucketList() {
   fetch('/data').then(response => response.json()).then((list) => {
     const listElement = document.getElementById('list-container');
     listElement.innerHTML = '';
     let bucketList = "";
 
-    for (let i = 0; i < list.length; i++) {
-      bucketList += (i + 1) + ": " + list[i] + '\n';
+    // Display elements in a numerical list
+    let count = 1;
+    for (i = 0; i < list.length; i++) {
+      if (list[i] === "") {
+        bucketList += "";
+      } 
+      else {
+        bucketList += count + ": " + list[i] + '\n';
```

Comment:
How about wrapping these in an `<ol>` with an `<li>` for each item and letting the browser take care of the numbering? Making each comment a separate HTML element would also add flexibility for future CSS styling and probably improve the screenreader exerience.